unresolved external symbol __progname

Hi,

I'm trying to build (branch `humble`) in win11 and VS2022 and I had a link error:
>rmw_init.obj : error LNK2001: unresolved external symbol __progname [C:\ros\humble\build\rmw_iceoryx_cpp\rmw_iceoryx_cpp.vcxproj]

the code is at:
https://github.com/ros2/rmw_iceoryx/blob/4134ce2c2ff8adfe2fb85cc67d644065e316f3bc/rmw_iceoryx_cpp/src/rmw_init.cpp#L98

I searched but the extern name of `__progname` is never found. Then in the whole source code of ROS2, I found 

https://github.com/ros2/rcutils/blob/025f78fa7c03666b5fa1057af00a32ed8bbd64a0/src/process.c#LL71C1-L74C6
```
#elif defined __QNXNTO__ || defined __OHOS__
  extern char * __progname;
  const char * appname = __progname;
#else
```

but I'm not sure if it's related.

--- a/rmw_iceoryx_cpp/src/rmw_init.cpp
+++ b/rmw_iceoryx_cpp/src/rmw_init.cpp
@@ -19,6 +19,7 @@
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 
 #include "rcutils/error_handling.h"
+#include "rcutils/process.h"
 
 #include "rmw/error_handling.h"
 #include "rmw/impl/cpp/macros.hpp"
@@ -95,8 +96,9 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
   context->impl = nullptr;
 
   // create a name for the process to register with the RouDi daemon
-  extern char * __progname;
-  auto progName = std::string(__progname);
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  auto progam_name = rcutils_get_executable_name(allocator);
+  auto progName = std::string(progam_name);
   /// @todo we could check with the introspection topics beforehand if the name is already used
   auto name = progName + "_" + std::to_string(getpid());
 

